src/cherrypick.rs

```diff
@@ -0,0 +1,70 @@
+use std::mem;
+
+use crate::build::CheckoutBuilder;
+use crate::merge::MergeOptions;
+use crate::raw;
+use std::ptr;
+
+/// Options to specify when cherry picking
+pub struct CherrypickOptions<'cb> {
+    mainline: u32,
+    checkout_builder: Option<CheckoutBuilder<'cb>>,
+    merge_opts: Option<MergeOptions>,
+}
+
+impl<'cb> CherrypickOptions<'cb> {
+    /// Creates a default set of cherrypick options
+    pub fn new() -> CherrypickOptions<'cb> {
+        CherrypickOptions {
+            mainline: 0,
+            checkout_builder: None,
+            merge_opts: None,
+        }
+    }
+
+    /// Set the mainline value
```

Comment:
I'm not really sure what this field is and the documentation here isn't too helpful, could this be expanded?

Reply:
Yes! I added the comment of the field from libgit2.